### PR TITLE
feat: delay gumlet script load and set config to generate placeholder

### PR DESCRIPTION
**Pull request by Harshith46**
# Description

- Add a placeholder for a hero-image component which can be configured from the bk config. tix: https://github.com/quintype/ace-planning/issues/92
- delay the load of gumlet script which can be configured from the bk config to improve the score and lcp. tix: https://github.com/quintype/ace-planning/issues/478

Dependencies: arrow

Documentation: will be updated

## Type of change

Please delete options that are not relevant.
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [ ] tested locally 
- [ ] verified on page speed insights

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/isomorphic/components/gumlet-script-generator.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import get from "lodash.get";
+import React, { useEffect } from "react";
+import { useSelector } from "react-redux";
+
+export const gumletScriptGenerator = () => {
+  const placeholderDelay = useSelector(state =>
+    get(state, ["qt", "config", "publisher-attributes", "placeholder_delay"], {})
+  );
+
```
Comment (Athira001) [inline app/isomorphic/components/gumlet-script-generator.js:9]: why are we moving this script from layout.ejs?  We can keep it in partials, right?

File: app/isomorphic/components/story-grid.m.css [hunk 2]
```diff
@@ -1,3 +1,7 @@
 .story-grid-item-image {
   width: 100%;
 }
+
+.placeholder {
+  background-color: lightgray;
```
Comment (Athira001) [inline app/isomorphic/components/story-grid.m.css:6]: variable?

File: app/isomorphic/components/story-templates/blank.js [hunk 3]
```diff
@@ -59,7 +61,10 @@ const BlankStoryTemplate = props => {
       <div styleName="wrapper">
         <WithLazy margin="20px">
           {() => (
-            <figure className="blank-story-image" styleName="qt-image-16x9">
+            <figure
+              className="blank-story-image"
+              styleName={`qt-image-16x9 ${showImagePlaceholder ? "placeholder" : ""}`}
```
Comment (ags1773) [inline app/isomorphic/components/story-templates/blank.js:66]: can use a variable to generate styleName string

File: app/isomorphic/components/gumlet-script-generator.js (hunk 1 above)
Comment (Harshith46) [reply app/isomorphic/components/gumlet-script-generator.js:9]: we are doing on timeout, not necessary to keep in layout if not required on server side

File: app/isomorphic/components/story-grid.m.css (hunk 2 above)
Comment (Harshith46) [reply app/isomorphic/components/story-grid.m.css:6]: we don't have background-color var

**Review by ags1773 — APPROVED** [review]

File: app/client/render.js [hunk 4]
```diff
@@ -32,8 +33,10 @@ export function preRenderApplication(store) {
 export function renderApplication(store) {
   const enableAds = get(store.getState(), ["qt", "config", "ads-config", "dfp_ads", "enable_ads"]);
   const pageType = get(store.getState(), ["qt", "pageType"], null);
+  const placeholderDelay = get(store.getState(), ["qt", "config", "publisher-attributes", "placeholder_delay"]);
```
Comment (Jeevan-Kishore) [inline app/client/render.js:36]: what if a string is inserted?

File: app/isomorphic/components/gumlet-script-generator.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  useEffect(() => {
+    setTimeout(function() {
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/gumlet-script-generator.js:11]: always clear timeouts!

File: app/isomorphic/components/gumlet-script-generator.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import get from "lodash.get";
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/gumlet-script-generator.js:1]:
```suggestion
import get from "lodash/get";
```

File: app/isomorphic/components/gumlet-script-generator.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    setTimeout(function() {
+      const script = document.createElement("script");
+      script.src = "https://cdn.gumlet.com/gumlet.js/2.0/gumlet.min.js";
+      const node = document.getElementsByTagName("script")[0];
+      script.setAttribute("defer", "defer");
+      node.parentNode.insertBefore(script, node);
+    }, placeholderDelay * 1000);
+  }, []);
+
+  return <div />;
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/gumlet-script-generator.js:20]: huh?

File: app/isomorphic/components/story-grid.m.css (hunk 2 above)
Comment (Jeevan-Kishore) [reply app/isomorphic/components/story-grid.m.css:6]: create one

**Review by Jeevan-Kishore — CHANGES_REQUESTED** [review]

File: app/isomorphic/components/gumlet-script-generator.js (hunk 1 above)
Comment (Athira001) [reply app/isomorphic/components/gumlet-script-generator.js:9]: Then how the images will load on ssr?

File: app/isomorphic/components/gumlet-script-generator.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import get from "lodash/get";
+import { useEffect } from "react";
+import { useSelector } from "react-redux";
+
+export const gumletScriptGenerator = () => {
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/gumlet-script-generator.js:5]: this as a component makes no sense, change to a function..

File: app/client/render.js (hunk 4 above)
Comment (Jeevan-Kishore) [reply app/client/render.js:36]: this?

**Review by Jeevan-Kishore — CHANGES_REQUESTED** [review]

**Pull request merged**